system/jlib/jstats.cpp

```diff
@@ -1988,19 +2087,71 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
                 cur.visit(visitor);
         }
     }
-
     virtual void visitChildren(IStatisticVisitor & visitor) const
     {
         for (auto const & cur : children)
             cur.visit(visitor);
     }
+    virtual void refreshAggregates(const StatisticsMapping & mapping, std::function<void(const char * scope, StatisticScopeType sst, StatisticKind kind, stat_type value)> & fWhenAggregateUpdated) override
+    {
+        if (isDirty)
+        {
+            CRuntimeStatisticCollection totals(mapping);
+            refreshAggregates(totals, fWhenAggregateUpdated);
+        }
+    }
+    virtual stat_type aggregateStatistic(StatisticKind kind) const override
+    {
+        stat_type sum;
+        if (!getStatisticSum(kind, sum)) // get sum of statistics at this level
+        {
+            // if no stats at this level, then get sum of stats from children
+            for (auto & child : children)
+                sum += child.aggregateStatistic(kind);
+        }
+        return sum;
+    }
+    virtual void recordStats(const StatisticsMapping & mapping, IStatisticCollection * sourceStatsCollection, std::initializer_list<const StatsScopeId> path) override
+    {
+        CStatisticCollection * curSrcCollection = static_cast<CStatisticCollection *>(sourceStatsCollection);
```

Comment:
not great to static_cast, can't be sure of what sourceStatsCollection is or will be in future.
I don't think it's necessary either, using the existing methods [ getNumStatistics and getStatistic(StatisticKind & kind, unsigned __int64 & value, unsigned idx) ], you can achieve the same reasonably efficiently (albeit for a few extra virtual calls)